internal/logging/panic_test.go

```diff
@@ -0,0 +1,47 @@
+package logging
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+)
+
+func TestPanicRecorder(t *testing.T) {
+	rec := panics.registerPlugin("test")
+
+	output := []string{
+		"panic: test",
+		"  stack info",
+	}
+
+	for _, line := range output {
+		rec(line)
+	}
+
+	expected := fmt.Sprintf(pluginPanicOutput, "test", strings.Join(output, "\n"))
+
+	res := PluginPanics()
+	if len(res) == 0 {
+		t.Fatal("no output")
+	}
+
+	if res[0] != expected {
+		t.Fatalf("expected: %q\ngot: %q", expected, res[0])
+	}
+}
+
+func TestPanicLimit(t *testing.T) {
+	rec := panics.registerPlugin("test")
+
+	rec("panic: test")
+
+	for i := 0; i < 200; i++ {
+		rec(fmt.Sprintf("LINE: %d", i))
+	}
+
+	res := PluginPanics()
+	// take the extra formatting into account
+	if len(res) > panics.maxLines+15 {
+		t.Fatalf("expected no more than %d lines, got: %d", panics.maxLines+50, len(res))
```

Comment:
Should this be `panics.maxLines+15`?

Reply:
It's longer here, because we're adding the extra output text to the stack trace. I'm going to make this check the actual message for the line count anyway, which will make it more obvious what it's doing.

Reply:
Ah, this test wasn't quite up to date with the code, fixed.